File: problem_set_2/valid_parentheses.py

def isValid(s: str) -> bool:
    prev_length = 0
    while len(s) != prev_length:
        s = s.replace("()", "").replace("[]", "").replace("{}", "")
        prev_length = len(s)
    return s == ""

if __name__ == "__main__":
    test_cases = [
        "()", "()[]{}", "(]", "([)]", "{[]}"
    ]
    for test in test_cases:
        print(isValid(test))

File: problem_set_2/valid_parentheses.go
package main

import (
	"fmt"
	"strings"
)

func isValid(s string) bool {
	for prev_length:=0; prev_length != len(s); prev_length = len(s){
		s = strings.ReplaceAll(strings.ReplaceAll(strings.ReplaceAll(s,"()",""),"[]",""),"{}","")
	}
	return s == ""
}

func main() {
	test_cases := []string{
        "()", "()[]{}", "(]", "([)]", "{[]}",
	}
	for _, test := range test_cases{
		result := isValid(test)
		fmt.Println(result)
	}
}
	